commonjs-extension-resolution-loader/loader.js

```diff
@@ -1,27 +1,45 @@
-import { existsSync } from 'fs';
-import { createRequire } from 'module';
+import { builtinModules } from 'node:module';
 import { dirname } from 'path';
-import { URL, fileURLToPath, pathToFileURL } from 'url';
+import { cwd } from 'process';
+import { fileURLToPath, pathToFileURL } from 'url';
+import { promisify } from 'util';
 
-const require = createRequire(import.meta.url);
-const baseURL = pathToFileURL(process.cwd() + '/').href;
+import resolveCallback from 'resolve/async.js';
 
-export function resolve(specifier, context, defaultResolve) {
+const resolveAsync = promisify(resolveCallback);
+
+const baseURL = pathToFileURL(cwd() + '/').href;
```

Comment:
TLDR: you might need to wrap the `cwd()` in a `try`/`catch`.

In `internal/process/esm_loader.js`, this cwd is wrapped in a try/catch. I don't remember why (I asked and Bradley answered, and I think I forgot document the reason in a comment).